src/components/Modal/SocialIcons.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect } from "react";
+import styled from "styled-components";
+const IconsContainer = styled.div`
+  display: flex;
+  gap: 32px;
+`;
+
+const IconContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
+  cursor: pointer;
+`;
+
+const IconImage = styled.img`
+  width: 42px;
+  height: 42px;
+`;
+
+const IconText = styled.span`
+  font-size: 13px;
+  font-weight: 400;
+  text-align: center;
+`;
+
+function SocialIcons({ id }: { id: number | null }) {
+  const shareUrl = `${window.location.origin}/shared/${id}`;
+
+  const { Kakao }: any = window;
+
+  useEffect(() => {
+    if (Kakao) {
+      Kakao.cleanup();
+      Kakao.init(process.env.REACT_APP_KAKAO_KEY);
+      window.kakao.isInitialized();
+    }
+  }, [Kakao]);
```

Comment:
c: 이 부분 추후 훅으로 분리하면 좋을 것 같아요

```suggestion
  const Kakao = useKakao();
```